core/src/main/java/org/apache/cxf/transport/common/gzip/GZIPOutInterceptor.java

```diff
@@ -337,7 +337,10 @@ private static void addHeader(Message message, String name, String value) {
         if (header.isEmpty()) {
             header.add(value);
         } else {
-            header.set(0, header.get(0) + "," + value);
+            String existingValue = header.get(0);
+            if (!existingValue.contains(value)) {
```

Comment:
I am wondering if this is even correct, why not to add values?
I believe we may just use ` header.add`, the multiple values are concatenated together using `,` (at least, should):

```
if (!header.contains(value)) {
    header.add(value);
}
```
```

Reply:
I was thinking the same, and I confirm, I get the same result by adding the item directly to the list: values are concatenated with a comma.